pkg/master/thirdparty/thirdparty.go

```diff
@@ -100,23 +117,42 @@ func (m *ThirdPartyResourceServer) HasThirdPartyResource(rsrc *extensions.ThirdP
 	path := extensionsrest.MakeThirdPartyPath(group)
 	m.thirdPartyResourcesLock.Lock()
 	defer m.thirdPartyResourcesLock.Unlock()
-	entry := m.thirdPartyResources[path]
-	if entry == nil {
+	tpg := m.thirdPartyResources[path]
+	if tpg == nil {
 		return false, nil
 	}
 	plural, _ := meta.KindToResource(unversioned.GroupVersionKind{
 		Group:   group,
 		Version: rsrc.Versions[0].Name,
 		Kind:    kind,
 	})
-	_, found := entry.storage[plural.Resource]
-	return found, nil
+	for _, version := range rsrc.Versions {
+		entry, _ := tpg.versionedEntry[version.Name]
+		if entry == nil {
+			return false, nil
+		}
+		_, found := entry.storage[plural.Resource]
+		if !found {
+			return false, nil
+		}
+	}
+	return true, nil
 }
 
 func (m *ThirdPartyResourceServer) removeThirdPartyStorage(path, resource string) error {
```

Comment:
how about change this to `removeThirdPartyStorage(group, version, resource string)`, wdyt?